logic/src/commonMain/kotlin/com/wire/kalium/logic/data/conversation/ConversationRepository.kt

```diff
@@ -453,17 +457,20 @@ internal class ConversationDataSource internal constructor(
         invalidateMembers: Boolean
     ) = wrapStorageRequest {
         val conversationEntities = conversations
-            .map { conversationResponse ->
-                conversationMapper.fromApiModelToDaoModel(
-                    conversationResponse,
-                    mlsGroupState = conversationResponse.groupId?.let {
-                        mlsGroupState(
-                            idMapper.fromGroupIDEntity(it),
-                            originatedFromEvent
-                        )
-                    },
-                    selfTeamIdProvider().getOrNull(),
-                )
+            .mapNotNull { conversationResponse ->
+                val mlsGroupState = conversationResponse.groupId?.let {
+                    mlsGroupState(idMapper.fromGroupIDEntity(it), originatedFromEvent)
+                }
+                // if group state is left, then we don't want to persist the conversation
+                if (mlsGroupState?.fold({ true }, { false }) == true) {
```

Comment:
Suggestion: Maybe we could extract this to a function, so we reuse it and documented 

```
fun shouldPersistMLSConversation() = mlsGroupState?.fold({ true }, { false }) == true
```

Reply:
Great idea!